GitUI/UserControls/RevisionGrid/Columns/MessageColumnProvider.cs

```diff
@@ -520,7 +521,15 @@ int RefTypeRank(IGitRef gitRef)
             }
         }
 
-        private static string[] GetCommitMessageLines(GitRevision revision) =>
-            (revision.Body?.Trim() ?? revision.Subject).Split(Delimiters.LineFeed, StringSplitOptions.RemoveEmptyEntries);
+        public override void Refresh(int rowHeight, in VisibleRowRange range)
+        {
+            _notesInSeparateColumn = AppSettings.ShowGitNotesColumn;
+            base.Refresh(rowHeight, in range);
+        }
+
+        private string GetText(GitRevision revision) => (_notesInSeparateColumn ? revision.Body : UIExtensions.FormatBodyAndNotes(revision.Body, revision.Notes)) ?? revision.Subject;
```

Comment:
```suggestion
        private string GetText(GitRevision revision) => (_notesInSeparateColumn ? revision.Body ?? revision.Subject: UIExtensions.FormatBodyAndNotes(revision.Body ?? revision.Subject, revision.Notes));
```

Reply:
Why inline this distinction between "body" and "subject" everywhere in the code? "Body" getter already does this. IMHO in does not need to check "HasMultilineChange". Do we really need null there in some (impossible) cases?